### PR TITLE
Fix an edge-case involving rollups and count(*)

### PR DESCRIPTION
**Pull request by rjmac**
See the new test in RollupExactTest for details

### CODE REVIEW

File: soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/rollup/RollupTestHelper.scala [hunk 1]
```diff
@@ -122,7 +122,7 @@ trait RollupTestHelper extends TestHelper { this: Assertions =>
   object TestRollupInfo {
     def apply(id: Int, name: String, tf: TableFinder[TestMT], soql: String): TestRollupInfo = {
       val Right(foundTables) = tf.findTables(0, soql, Map.empty)
-      val analysis = analyzer(foundTables, UserParameters.empty) match {
+      val analysis = systemColumnPreservingAnalyzer(foundTables, UserParameters.empty) match {
```
Comment (rjmac) [inline soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/rollup/RollupTestHelper.scala:125]: Oops, didn't mean to check this line in; it's not necessary for the repro.

File: soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/rollup/RollupExactTest.scala [hunk 2]
```diff
@@ -379,4 +379,28 @@ class RollupExactTest extends FunSuite with MustMatchers with RollupTestHelper w
       TestRollupExact(select, rollup, analysis.labelProvider) must be (None)
     }
   }
+
+  test("parameterless aggregate one-to-one rewrite bug") {
```
Comment (rlvoyer) [inline soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/rollup/RollupExactTest.scala:383]: Nit: I think it would helpful as a reader of this code if the spec description actually called out desired behavior rather than referring to the bug.

Something like this?
```scala
test("rollup on an an aggregate with a parameter cannot be applied to a query with the same groupby clause but a different aggregate and parameter")
```

Comment (rjmac) [reply soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/rollup/RollupExactTest.scala:383]: That's not quite the bug though; it was specifically the fact that "count(*)" has no parameters that was the trigger.  It is indeed not an ideal name though.

**Review by dnabilali — APPROVED** [review]

File: soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/rollup/RollupExactTest.scala [hunk 3]
```diff
@@ -379,4 +379,28 @@ class RollupExactTest extends FunSuite with MustMatchers with RollupTestHelper w
       TestRollupExact(select, rollup, analysis.labelProvider) must be (None)
     }
   }
+
+  test("A parameterless aggregate cannot be used with a rollup that has the same grouping which does not produce that aggregate as output") {
```
Comment (rlvoyer) [inline soql-analyzer/src/test/scala/com/socrata/soql/analyzer2/rollup/RollupExactTest.scala:383]: 👍

**Review by saraid — APPROVED** [review]
I think it makes sense to me.
